src/promptflow-azure/promptflow/azure/_storage/cosmosdb/span.py

```diff
@@ -28,17 +29,17 @@ class Span:
 
     def __init__(self, span: SpanEntity, collection_id: str, created_by: dict) -> None:
         self.name = span.name
-        self.context = span._content[SpanFieldName.CONTEXT]
-        self.kind = span._content[SpanFieldName.KIND]
-        self.parent_id = span.parent_span_id
-        self.start_time = span._content[SpanFieldName.START_TIME]
-        self.end_time = span._content[SpanFieldName.END_TIME]
-        self.status = span._content[SpanFieldName.STATUS]
-        self.attributes = span._content[SpanFieldName.ATTRIBUTES]
-        self.events = span._content[SpanFieldName.EVENTS]
-        self.links = span._content[SpanFieldName.LINKS]
-        self.resource = span._content[SpanFieldName.RESOURCE]
-        self.partition_key = span.session_id
+        self.context = span.context
+        self.kind = span.kind
+        self.parent_id = span.parent_id
+        self.start_time = span.start_time.isoformat()
+        self.end_time = span.end_time.isoformat()
```

Comment:
Could you give a example of before/after do "isoformant"? And I not sure if it should be called when init SpanEnitity.

Reply:
in previous span entity, `start_time` and `end_time` should be already a string; while in new entity, this will be a `datetime.datetime` object, which looks like `datetime.datetime(2024, 3, 28, 7, 11, 31, 691990)`, and `.isoformat()` will return a string in ISO 8601 standard, which looks like `2024-03-28T07:15:14.093111`, which in my understanding, it's the same thing in the previous entity.

![image](https://github.com/microsoft/promptflow/assets/38847871/4a1f95d9-9382-404a-93a7-79fce935b71e)